src/icalendar.ml

```diff
@@ -696,8 +696,16 @@ module Writer = struct
   let dates_or_times_or_periods_to_ics dt buf =
     let swap f a b = f b a in
     match dt with
-    | `Dates xs -> List.iter (date_to_ics buf) xs
-    | `Datetimes xs -> List.iter (swap timestamp_to_ics buf) xs
+    | `Dates xs -> (match xs with
+        | _ -> date_to_ics buf (List.hd xs);
```

Comment:
Using `List.hd` is generally a code smell in OCaml. This is going to crash if `xs` is the empty list.

You should pattern-match on it to be sure you are handling this case properly (and thus, you won't need `List.hd` and `List.tl` anymore):

```ocaml
  match xs with
  | [] -> ...
  | hd :: tl -> ...
```

Reply:
Thank you for this. The pattern matching would be great! Will fix in a moment.

Reply:
Fixed and submitted.